### DISCUSSION TITLE
[Explore] Ability to turn a relative time range into an absolute one

### DISCUSSION THREAD

**Discussion by owen-d** [grafana/grafana#37893] [Explore]
I would _love_ the ability to turn a relative window into an absolute one (for persisting historical context when posting to slack, etc) somewhere in here. This would be a lifesaver when debugging an incident and I want the postmortem to retain relevant context.

![image](https://user-images.githubusercontent.com/8173478/129386047-9b2a78bd-cd55-40f5-8fcf-13dcbfc0a042.png)

**Comment by sakjur** [comment]
:+1: That reminds me of GitHub's superuseful `y` keyboard shortcut which allows you to get the canonical form of a URL. I think it'd be nice to have at least a similar keyboard shortcut that returns the URL with relative time converted into absolute time.

> `y` | Expand a URL to its canonical form. For more information, see "Getting permanent links to files."

from https://docs.github.com/en/get-started/using-github/keyboard-shortcuts

**Comment by aocenas** [comment]
This feels like something that would make sense in general (sharing a dashboard etc) and the timepicker isn't really special for explore. @hugohaggmark what do you think? It seems like this would be more appropriate for User Essentials.

**Reply by hugohaggmark** [reply]
Hmm, actually I think @grafana/grafana-bi-squad is working on the Timepicker.

**Reply by oscarkilhed** [reply]
We've pretty much ended that work. There is an ongoing effort to improve the UX for the timepicker though. This should probably be explored in that context.

**Reply by ivanahuckova** [reply]
+cc @nadinevehling (I couldn't find Niat), but I know they are currently working on time picker improvements.

**Comment by davkal** [comment]
I've picked this up in #43802

**Comment by aocenas** [comment]
Moved this to the general category, which seems more appropriate.

**Comment by ivanahuckova** [comment]
I think we can close this one as it was implemented and merged https://github.com/grafana/grafana/pull/43802. I am wondering if we just lock this conversation or if we should turn this into issue and then close that issue?

**Reply by aocenas** [reply]
Would say turning this to issue and closing it but that will still leave this discussion so still need to be locked and marked as discussion/resolved. In any case, I thought maybe somebody would argue to have it as a button or something more discoverable. Or at least @owen-d can say if he is happy with the shortcut.

**Comment by AustinDeric** [comment]
this is a great product feature request. yes please.

**Comment by ivanahuckova** [comment]
@AustinDeric this is possible with shortcut `t a` that makes relative time range absolute 
<img width="1912" alt="image" src="https://user-images.githubusercontent.com/30407135/222729827-e75344e2-8f62-4662-8fda-d3fbf4d22b27.png">
